python/lsst/ts/Dome/configuration_limits.py

```diff
@@ -0,0 +1,34 @@
+from astropy import units as u
+import math
+
+
+configuration_limits = {
+    "AMCS": {
+        "jmax": {
+            "description": "AMCS Maximum Jerk",
+            "upper_limit": (3.0 * math.pi / 180.0) * u.rad / u.s ** 3,
```

Comment:
Looking at `config_lls` I don't think you are getting any value out of these being astropy quantities. It seems to me that what you want to do is easily convert between degrees and radians, while ignoring the time component, and astropy just doesn't seem to support that.

Please consider just using radians or degrees here. Personally I suggest degrees, for ease of use, but pick one and stick to it. You can use one code comment to document your convention (that seems less bulky than a separate units string per entry, which would be hard to enforce).

Also please consider a constant such as `RADIANS_PER_DEGREES`. It may be a few more characters but I think `3.0 * RADIANS_PER_DEGREES` is easier to read and the intent is clearer.

Reply:
Please consider replacing the dicts with a small struct-like custom class. Advantages:
* You need to check that fields are present; they are sure to be present.
* You can document the constructor parameters -- including the units.
* There is no need for the "description" and "upper_limit" fields -- the parameter documentation does all that for you.
* You can add a method to validate a dict of proposed values (e.g. in config_lls).
* You can even add a name field (that you could set to "AMCS" or "LWSCS"). I doubt you would find it helpful, but it might come in handy for error messages.

Then all this code becomes:
```
configuration_limits = dict(
    AMCS=ConfigurationLimit(jmax=3.0, amax=0.5, vmax=1.5),
    LWSCS=ConfigurationLimit(jmax=3.5, amax=0.875, vmax=1.75)
)
```
with the class defined somewhere. By the time you define the class there may be more net code, but I think the result will be significantly easier to read and use. Something along the line of:
```
class ConfigurationLimit:
    """...docs including the crucial Parameters section..."""
    def __init__(self, jmax, amax, vmax):
        self.jmax = jmax
        self.amax = amax
        self.vmax = vmax

   def check_config_data(self, data):
        if data["vmax"] > self.vmax * RADIANS_PER_DEGREE:
            raise ValueError(f"...")
```

Reply:
I overhauled this code completely based on the Slack chats we had today. Many thanks for all your inputs!